platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/pax/web/jetty/DelegatingHttpFilterHandler.java

```diff
@@ -0,0 +1,89 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.pax.web.jetty;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.util.Objects;
+import javax.servlet.Filter;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.codice.ddf.platform.filter.HttpFilter;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.handler.HandlerWrapper;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.util.tracker.ServiceTracker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Hands the request off to a set chain of servlet {@link Filter}s. Since SecurityFilters are run on
+ * each request, this provides a mechanism to add global servlet filters. As of OSGi R6, there is a
+ * proper way to define global servlets/filters/listeners/etc., defined by the HTTP Whiteboard spec.
+ * However, pax-web does not yet implement that feature, so we're left using this workaround.
+ *
+ * <p>When https://ops4j1.jira.com/browse/PAXWEB-1123 is resolved, this workaround should be
+ * revisited.
+ */
+public class DelegatingHttpFilterHandler extends HandlerWrapper {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DelegatingHttpFilterHandler.class);
+
+  private final ServiceTracker<HttpFilter, HttpFilter> filterTracker;
+
+  @VisibleForTesting
+  DelegatingHttpFilterHandler(ServiceTracker<HttpFilter, HttpFilter> filterTracker) {
+    this.filterTracker = Objects.requireNonNull(filterTracker);
+  }
+
+  public DelegatingHttpFilterHandler() {
+    Bundle bundle = FrameworkUtil.getBundle(DelegatingHttpFilterHandler.class);
+    Objects.requireNonNull(bundle, "Bundle must not be null");
+    Objects.requireNonNull(bundle.getBundleContext(), "Bundle has no valid BundleContext");
+
+    this.filterTracker =
+        new ServiceTracker(bundle.getBundleContext(), HttpFilter.class.getName(), null);
+  }
+
+  @Override
+  public void handle(
+      String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    HttpFilter[] filters = getFilters();
+    LOGGER.trace("Delegating to {} HttpFilters.", filters.length);
+
+    ProxyHttpFilterChain filterChain =
+        new ProxyHttpFilterChain(filters, getHandler(), target, baseRequest);
+    filterChain.doFilter(request, response);
+  }
+
+  private HttpFilter[] getFilters() {
+    HttpFilter[] filters = new HttpFilter[filterTracker.size()];
+    return filterTracker.getServices(filters);
```

Comment:
This is getting injected into the Jetty Handler chain via the jetty.xml config, so I couldn't just inject a service reference list like it did before. If anybody has better ideas for how to do this, I'm open to suggestions.